oio/event/filters/account_update.py

```diff
@@ -77,13 +86,28 @@ def process(self, env, cb):
             m2_services = [x for x in new_services if x.get('type') == 'meta2']
             if not m2_services:
                 # No service in charge, container has been deleted
-                self.account.container_update(
+                try:
+                    self.account.container_update(
                         url.get('account'), url.get('user'),
                         {'dtime': event.when / 1000000.0},
-                        read_timeout=ACCOUNT_TIMEOUT)
+                        connection_timeout=self.connection_timeout,
+                        read_timeout=self.read_timeout)
+                except OioTimeout as exc:
+                    msg = 'failed to remove %s from account %s: %s' % (
+                        url.get('user'), url.get('account'), str(exc))
+                    resp = EventError(event=event, body=msg)
+                    return resp(env, cb)
             else:
-                self.account.account_create(
-                    url.get('account'), read_timeout=ACCOUNT_TIMEOUT)
+                try:
+                    self.account.account_create(
+                        url.get('account'),
+                        connection_timeout=self.connection_timeout,
+                        read_timeout=self.read_timeout)
+                except OioTimeout as exc:
+                    # The account will be autocreated by the next event,
+                    # just warn and continue.
+                    self.logger.warn('Failed to create account %s: %s',
```

Comment:
Which events will autocreate the account if this fail ?

Reply:
I am thinking of `storage.container.state` events.

Reply:
Example with `account.services` events disabled:
```
fvennetier@linux-cgzg:~/build/public_git/oio-sds> grep account.services ~/.oio/sds/conf/event-handlers-1.conf -A 2
[handler:account.services]
#pipeline = account_update volume_index
pipeline = volume_index
fvennetier@linux-cgzg:~/build/public_git/oio-sds> openio account list
+-----------+
| Name      |
+-----------+
| myaccount |
+-----------+
fvennetier@linux-cgzg:~/build/public_git/oio-sds> openio container create fve --oio-account myaccount2
+------+---------+
| Name | Created |
+------+---------+
| fve  | True    |
+------+---------+
fvennetier@linux-cgzg:~/build/public_git/oio-sds> openio account list
+------------+
| Name       |
+------------+
| myaccount  |
| myaccount2 |
+------------+
fvennetier@linux-cgzg:~/build/public_git/oio-sds> 
```